layout/src/lib.rs

```diff
@@ -1,9 +1,45 @@
-extern crate cassowary;
-extern crate euclid;
+// ---- START CLIPPY CONFIG
+
+#![cfg_attr(all(not(test), feature="clippy"), warn(result_unwrap_used))]
+#![cfg_attr(feature="clippy", warn(unseparated_literal_suffix))]
+#![cfg_attr(feature="clippy", warn(wrong_pub_self_convention))]
+
+// Enable clippy if our Cargo.toml file asked us to do so.
+#![cfg_attr(feature="clippy", feature(plugin))]
+#![cfg_attr(feature="clippy", plugin(clippy))]
+
+#![warn(missing_copy_implementations,
+        trivial_numeric_casts,
+        trivial_casts,
+        unused_extern_crates,
+        unused_import_braces,
+        unused_qualifications)]
+#![cfg_attr(feature="clippy", warn(cast_possible_truncation))]
+#![cfg_attr(feature="clippy", warn(cast_possible_wrap))]
+#![cfg_attr(feature="clippy", warn(cast_precision_loss))]
+#![cfg_attr(feature="clippy", warn(cast_sign_loss))]
+#![cfg_attr(feature="clippy", warn(missing_docs_in_private_items))]
+#![cfg_attr(feature="clippy", warn(mut_mut))]
+
+// Disallow `println!`. Use `debug!` for debug output
+// (which is provided by the `log` crate).
+#![cfg_attr(feature="clippy", warn(print_stdout))]
+
+// This allows us to use `unwrap` on `Option` values (because doing makes
+// working with Regex matches much nicer) and when compiling in test mode
+// (because using it in tests is idiomatic).
+#![cfg_attr(all(not(test), feature="clippy"), warn(result_unwrap_used))]
+#![cfg_attr(feature="clippy", warn(unseparated_literal_suffix))]
+#![cfg_attr(feature="clippy", warn(wrong_pub_self_convention))]
+
+// ---- END CLIPPY CONFIG
```

Comment:
I was wondering what's the reason for using this particular set of clippy settings and not using the defaults? I assume the clippy defaults are what they are in order to minimize false positives, and there is a `clippy_pedantic` setting which is closer to this list.
To be honest at this stage it feels like extra strict linting could be more trouble than it's worth, if anything I'd be more inclined to make it less strict, in order to minimize the use of `#[cfg_attr..allow..]`.

Reply:
The lints were mostly taken from here: http://blog.faraday.io/good-strict-default-warnings-for-rust-code/

I mostly wanted to make clippy warnings a bit stricter than the default. Sure, you can't prevent everything, but I'd much rather have linting than let potential errors slide by. These lints only get activated when `cargo clippy` is run, they don't impact the regular build process.

I also wouldn't run `cargo clippy` on each compilation, but rather maybe every few weeks. I'll try to remove the `#[cfg_attr..allow..]` things.

Reply:
Ok hmm.. they don't seem that unreasonable. I do wonder why these aren't clippy defaults if they do show few false positives though. My preference is that if linting is done, it's checked on every PR by travis, so that issues are caught before they are merged, but it's something I only want to spend time focusing on after limn is in a more useful state.

I want to stress I still think it's better to think of limn as a prototype right now, so that said, I'm ok merging these clippy settings but I'll probably hold off on enforcing them or paying them much attention to them until limn is more mature.